kms-provider-kroxylicious-inmemory/pom.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright Kroxylicious Authors.
+
+    Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.kroxylicious</groupId>
+        <artifactId>kroxylicious-parent</artifactId>
+        <version>0.4.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>kms-provider-kroxylicious-inmemory</artifactId>
```

Comment:
It's worth remembering that maven itself doesn't care: A groupId is just a namespace in which artifactIds live. We need to be careful to avoid a categorisation that's arbitrary and prone to bikeshedding about how to [lump and split](https://xkcd.com/2518/).

So who is using these groupIds, and what they might usefully infer from them? Obviously _we_ use them, but we can be expected to get by without needing to infer anything about them: We (should) know what all the artifacts are. The real users of groupId are Java programmers naming them as dependencies, for example in their filters, in their KMS providers, or any other plugin SPI we come up with. 

So perhaps there's a case for `io.kroxylicious.api` for holding all things which which we consider public APIs (see whatever other comment I made earlier today about how we could end up with multiple of those). And `io.kroxylicious.impl` for the runtime. That makes it much clearer to a developer what they ought (not) to be depending on. Possibly it helps us too wrt BOMs. One thing that's not clear in this scheme is where service/plugin implementations (e.g. kms providers) live. They're mostly impl, but their config class is public API. So maybe there's a case for `io.kroxylicious.spi`. For example:

* `io.kroxylicious.api:filter`
* `io.kroxylicious.spi:kms`
* `io.kroxylicious.provider:kms-provider-kroxylicious-inmemory`
* `io.kroxylicious.provider:kms-provider-aws-kms`
* `io.kroxylicious.provider:filter-encryption`
* `io.kroxylicious.provider:filter-multitenant`
* `io.kroxylicious.impl:runtime`
* `io.kroxylicious.test:unit-test-support`
* `io.kroxylicious.test:integration-test-support`

Under these scheme:
* A filter author would expect to depend on modules exclusively in the `api` group (in `provided` scope) and the `spi` group (whose scope might depends on what we do about classloading). And `io.kroxylicious.test` for support artifacts (which should be in `test` scope). 
* Someone putting together a proxy application (writing a proxy configuration file and putting jars in the classpath) doesn't directly care about GAV, but does care about jar file names. In any case they're dealing exclusively with adding jars from the `provider` group to a classpath that already includes the `impl` group. 

The rules for naming would be:

* Anything in `impl` could only depend on `api`
* A filter must be in `provider`.
* Anything in `provider` can only depend on `api`, and `spi`

We'd probably want to pull the plugin package into its own API so that providers can use it without having to depend on `api:filter`.

Or something like that. Or maybe this is all better done at the java module level, if we ever go there. 

Anyway, I think this is sufficiently unclear that it shouldn't block this PR.